distribution/docker/build.gradle

```diff
@@ -91,20 +98,79 @@ void addCopyDockerfileTask(final boolean oss) {
   }
 }
 
+Closure createAndSetWritable = {location ->
+  File file = file(location);
+  file.mkdirs()
+  file.setWritable(true, false)
+}
+
+task copyKeystore(type: Sync) {
+  from project(':x-pack:plugin:core')
+          .file('src/test/resources/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode.jks')
+  into "${buildDir}/certs"
+  doLast {
+    file("${buildDir}/certs").setReadable(true, false)
+    file("${buildDir}/certs/testnode.jks").setReadable(true, false)
+  }
+}
+
 preProcessFixture {
   dependsOn taskName("copy", true, "DockerContext")
   dependsOn taskName("copy", true, "Dockerfile")
   dependsOn taskName("copy", false, "DockerContext")
   dependsOn taskName("copy", false, "Dockerfile")
+  dependsOn copyKeystore
+  doLast {
+    // tests expect to have an empty repo
+    project.delete("${buildDir}/repo")
+    project.delete("${buildDir}/oss-repo")
+    createAndSetWritable("${buildDir}/repo")
+    createAndSetWritable("${buildDir}/oss-repo")
+    createAndSetWritable("${buildDir}/logs/default-1")
+    createAndSetWritable("${buildDir}/logs/default-2")
+    createAndSetWritable("${buildDir}/logs/oss-1")
+    createAndSetWritable("${buildDir}/logs/oss-2")
+  }
 }
 
-postProcessFixture.doLast {
-  println "docker default distro is on port: ${ext."test.fixtures.elasticsearch-default.tcp.9200"}, " +
-          "oss is on: ${ext."test.fixtures.elasticsearch-oss.tcp.9200"}"
+configurations {
+  restSpec
+}
+dependencies {
+  restSpec "org.elasticsearch:rest-api-spec:${version}"
+}
+
+task copyRestTests(type: Sync) {
```

Comment:
Rather than define a new task here, we should just configure the existing `processTestResources` task to do this. Its job is already to copy stuff into the resources output directory and can be configured like any other `CopySpec`.

Reply:
I specifically wanted to have a  `Sync` task to make sure that there are no tests accidentally left in the resources directory when switching branches etc.

Reply:
The `ProcessResources` task already does this.

https://github.com/gradle/gradle/blob/master/subprojects/language-jvm/src/main/java/org/gradle/language/jvm/tasks/ProcessResources.java#L30